### PR TITLE
fix(addon-commerce): emulate focus on iOS by `tuiAutoFocus`

### LINKED ISSUE #3775: 🐞 - Mobile keyboard overlaps dialog
### Which @taiga-ui/* package(s) are the source of the bug?

core

### Please provide a link to a minimal reproduction of the bug

https://github.com/viktorzdanovich/taiga-ui-issues

### Is this issue blocking you?

Blocking

### Description

Steps to reproduce:
1) Tap "Pay" button
2) Tap "New card" button
3) Tap somewhere outside to dismiss focused field and hide the keyboard
4) Tap the card field

<details>
  <summary>Demo</summary>

  
Simulator Screen Recording - iPhone 14 Pro Max - 2023-02-28 at 13 51 00

</details>

### Angular version

14.2.0

### Taiga UI version

3.19.0

### Which browsers have you used?

- [ ] Chrome
- [ ] Firefox
- [X] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [ ] Windows
- [ ] Linux
- [X] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

ios-keyboard

Closes #3775 #3771 #3792

## What is the new behavior?

https://user-images.githubusercontent.com/12021443/222735859-a7ff7ab4-30ac-464c-a8b9-cde6cb550fb2.mp4

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by splincode** [comment]
@waterplea @nsbarsukov @vladimirpotekhin Hello, sorry for ping, could you please review me?

File: projects/cdk/directives/auto-focus/autofocus.directive.ts [hunk 1]
```diff
@@ -10,18 +10,29 @@ import {
 @Directive({
     selector: '[tuiAutoFocus]',
     providers: TUI_AUTOFOCUS_PROVIDERS,
+    exportAs: 'tuiAutoFocus',
 })
-export class TuiAutoFocusDirective implements AfterViewInit {
+export class TuiAutoFocusDirective<T extends HTMLElement = HTMLElement>
+    extends ElementRef<T>
```
Comment (nsbarsukov) [inline projects/cdk/directives/auto-focus/autofocus.directive.ts:16]: Why do we need to extend this class ?

File: projects/demo/src/modules/components/dialog/examples/9/index.less [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    padding-top: 1.25rem;
+}
+
+.pay-button {
+    margin-top: 1rem;
+    width: 100%;
+}
+
+.icon-group {
+    background: #f6f6f6;
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/dialog/examples/9/index.less:14]: `var(--tui-base-02)` ?

File: projects/demo/src/modules/components/dialog/examples/9/pay-modal/pay-modal.component.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                () => {
+                    this.setPayProcessing(false);
+                    this.context.$implicit.complete();
+                },
+                () => this.setPayProcessing(false),
+            );
+    }
+
+    deleteCards(): void {
+        // noop
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/dialog/examples/9/pay-modal/pay-modal.component.ts:126]: ?

File: projects/demo/src/modules/components/dialog/examples/9/pay-modal/pay-modal.component.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                (card: AccountCard) => card.cardId !== this.selectedCardId,
+            );
+
+            this.focusCVC();
+        } else {
+            this.form.patchValue({card: null});
+            this.form.controls.card.addValidators(tuiCardNumberValidator);
+            this.selectedCardId = null;
+            this.savedCardsWithoutSelected = this.savedCards;
+            this.paymentMode = PaymentMode.ByNewCard;
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/dialog/examples/9/pay-modal/pay-modal.component.ts:93]: **nitpicking:**
It is a cool example, but it is little bit overcomplicated for documentation.

Can we simplify it ?

File: projects/demo/src/modules/components/dialog/examples/9/index.less (hunk 2 above)
Comment (splincode) [reply projects/demo/src/modules/components/dialog/examples/9/index.less:14]: But in dark mode I want only `#f6f6f6`

File: projects/cdk/directives/auto-focus/autofocus.directive.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/directives/auto-focus/autofocus.directive.ts:16]: for compatibility

```html
<input
  #myInput 
  placeholder="etc"
  tuiAutofocus
/>
```

after 

```html
<input
  #myInput="tuiAutofocus"
  placeholder="etc"
  tuiAutofocus
/>
```

```ts
@ViewChild (`myInput`) elementRef: ElementRef<HTMLInputElement> & AutofocusDirective;
```

File: projects/demo/src/modules/components/dialog/examples/9/pay-modal/pay-modal.component.ts (hunk 4 above)
Comment (splincode) [reply projects/demo/src/modules/components/dialog/examples/9/pay-modal/pay-modal.component.ts:93]: fix

File: projects/demo/src/modules/components/dialog/examples/9/pay-modal/pay-modal.component.ts (hunk 3 above)
Comment (splincode) [reply projects/demo/src/modules/components/dialog/examples/9/pay-modal/pay-modal.component.ts:126]: fix

**Comment by splincode** [comment]
@nsbarsukov @vladimirpotekhin hello, please review me

File: projects/cdk/directives/auto-focus/autofocus.directive.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/directives/auto-focus/autofocus.directive.ts]: fix

**Review by vladimirpotekhin — APPROVED** [review]

**Comment by splincode** [comment]
Link: https://github.com/Tinkoff/taiga-ui/issues/3792

image

**Review by nsbarsukov — APPROVED** [review]

File: projects/cdk/directives/auto-focus/autofocus.directive.ts [hunk 5]
```diff
@@ -10,18 +10,30 @@ import {
...
+{
     @Input('tuiAutoFocus')
     autoFocus: boolean | '' = true;
 
     constructor(
+        @Inject(ElementRef) private readonly elementRef: ElementRef<T>,
         @Inject(TUI_AUTOFOCUS_HANDLER) private readonly handler: TuiAutofocusHandler,
     ) {}
 
+    get nativeElement(): T {
```
Comment (waterplea) [inline projects/cdk/directives/auto-focus/autofocus.directive.ts:26]: I don't like expanding public API unnecessarily for the sake of one component. Let's add `read: ElementRef` counterparts in `InputCardGroup` if we need native elements there. `exportAs` also doesn't seem to be required.

**Review by waterplea — CHANGES_REQUESTED** [review]

File: projects/cdk/directives/auto-focus/autofocus.directive.ts (hunk 5 above)
Comment (splincode) [reply projects/cdk/directives/auto-focus/autofocus.directive.ts]: fixed

**Review by waterplea — APPROVED** [review]

**Pull request merged**